pkg/controllers/dnspolicy/dnspolicy_controller.go

```diff
@@ -20,37 +20,51 @@ import (
 	"context"
 
 	"k8s.io/apimachinery/pkg/api/equality"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
 	ctrl "sigs.k8s.io/controller-runtime"
 	"sigs.k8s.io/controller-runtime/pkg/client"
-	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	crlog "sigs.k8s.io/controller-runtime/pkg/log"
 	"sigs.k8s.io/controller-runtime/pkg/source"
+	gatewayapiv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
 
 	"github.com/Kuadrant/multicluster-gateway-controller/pkg/apis/v1alpha1"
 	"github.com/Kuadrant/multicluster-gateway-controller/pkg/controllers/gateway"
 	"github.com/Kuadrant/multicluster-gateway-controller/pkg/dns"
+	"github.com/kuadrant/kuadrant-operator/pkg/reconcilers"
 )
 
 const (
-	DNSPolicyFinalizer = "kuadrant.io/dns-policy"
+	DNSPolicyFinalizer           = "kuadrant.io/dns-policy"
+	DNSPoliciesBackRefAnnotation = "kuadrant.io/dnspolicies"
 )
 
+type DNSPolicyRefsConfig struct{}
+
+func (c *DNSPolicyRefsConfig) PolicyRefsAnnotation() string {
+	return DNSPoliciesBackRefAnnotation
+}
+
 // DNSPolicyReconciler reconciles a DNSPolicy object
 type DNSPolicyReconciler struct {
-	Client client.Client
-
+	reconcilers.TargetRefReconciler
 	DNSProvider dns.Provider
 	HostService gateway.HostService
+	Placement   gateway.GatewayPlacer
```

Comment:
I see the use in the Gateway reconciler. Is adding to the DNSPolicy reconciler preparation for a next step maybe?

Reply:
It's used in the DNSPolicy to determine what what clusters a Gateway is actually placed on (OCM thing for multi cluster support). When we create a DNSRecord and add endpoints to it, we only use gateway addresses for gateways actually placed on workload clusters.

It would be nice though if this reconciler could work in a multi cluster (OCM) and singe cluster (non OCM) fashion. The whole DNSPolicy, much like RateLimitPolicy and AuthPolicy, could be effuse in it's own right outside of an OCM environment.

Reply:
We have an issue to aggregate the status of spoke gateways back to the hub gateway. The addresses will be part of that so we can prob disconnect this from the placement logic when that is in place